resources/design-document.md

```diff
@@ -0,0 +1,217 @@
+# Capstone Design Document
+
+---
+
+## 1. Problem Statement
+
+---
+
+People have to balance a lot of tasks when they cook. It's difficult for some to manually adjust recipes' ingredients to the amount they need versus the amount the recipe calls for. The goal of this app is to allow a user to build their own recipes and to provide them with the ability to easily adjust measurements according to the number of servings, switch between units of measurement, and create editable grocery lists based on the recipe's requirements.
+
+## 2. Top Questions to Resolve in Review
+
+---
+
+1. Would it be a good idea to cache the recipes that are called from the database?
+2. Does it make sense to store the ShoppingList items as a Map<Ingredient, Integer(quantity)> as opposed to 
+just having a Set<String(ingredientNames)>
+
+## 3. Use Cases
+
+---
+
+U1. As a customer, I want to create a new recipe
+
+U2. As a customer, I want to retrieve a specific recipe
+
+U3. As a customer, I want to be able to update a recipe with new values
+
+U4. As a customer, I want to create a list of ingredients that I can delete individual items from.
+
+U5. As a customer, I want variable units of measurement
+
+U6. As a customer, I want to be able to change amount of ingredients based on how many servings I want
+
+U7. As a customer, I want to sort recipes based on a certain ingredient
+
+U8. As a customer, I want to schedule meals ahead of time
```

Comment:
This was actually part of my original idea. I first kind of envisioned it as more of an organizational app, plotting meals on a weekly schedule, adding their ingredients and quantities needed to a shopping list. But the further I got in my design, it started feeling like an outlier. I just wanted to keep the idea somewhere in case I had time to implement it in a way that made sense.